operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/DefaultEventSourceManager.java

```diff
@@ -17,9 +17,8 @@
   private static final Logger log = LoggerFactory.getLogger(DefaultEventSourceManager.class);
 
   private final ReentrantLock lock = new ReentrantLock();
-  private Map<String, EventSource> eventSources = new ConcurrentHashMap<>();
-  private CustomResourceEventSource customResourceEventSource;
```

Comment:
Do you see it correctly that this has been moved to the Operator?  Why is that?
The intention was that this class will own all the custom resources related to a controller.
But are there particular reasons to move it? 
The downside is that if we want to have other events,between event sources and controllers vs event source, but also access access CustomResourceEventSource (where we thinking to move the caching for example) will not be accessible in this manager.

Reply:
The reason was that the method:

```
public void registerCustomResourceEventSource(
      CustomResourceEventSource customResourceEventSource) {
    this.customResourceEventSource = customResourceEventSource;
    this.customResourceEventSource.addedToEventManager();
}
```

Was not really meaningful as yes the `customResourceEventSource` was assigned to an instance of the DefaultEventManager but was never accesses by the manager. A better option would have been either to use `registerEventSource` as any other event source or pass the `customResourceEventSource` as a constructor parameter to make it clear that the `DefaultEventManager` requires it whereas as today, it looks like an unrelated method.

Reply:
I see, ok, yes this part definetelly needs improvements and refactorings. 
As I mentioned I would like to keep the event sources in one place, I think the constructor would make sense. 
CustomResourceEventSource currently has quite special position, it's required always as an event source, so constructor is good ide also do express this.

Reply:
I see, ok, yes this part definetelly needs improvements and refactorings. (Just like the others :) )
As I mentioned I would like to keep the event sources in one place, I think the constructor would make sense.
CustomResourceEventSource currently has quite special position, it's required always as an event source, so constructor is good ide also do express this.

Reply:
Note that I will probably make some other improvements after this, getting back to work on #392

Reply:
@csviri yep I think I got the reasoning and I do agree but for the sake of the lifecycle it was quite confusing so I opted to make at little bit more clear (i.e to have set-up / tear-down on the same side, in this case on the Operator). If we move it on the contructor then we can move its set-up/tear-down as part of the event manager.